libraries/AP_LandingGear/AP_LandingGear.cpp

```diff
@@ -208,6 +233,30 @@ void AP_LandingGear::update()
         
         gear_state_current = gear_state_new;
     }
+
+    /*
+      check for height based triggering
+     */
+    int16_t alt_m = constrain_int16(height_above_ground_m, 0, INT16_MAX);
```

Comment:
Why pass in a float that is then constrained to an int16? And since the minimum is zero, why not use uint16?

Reply:
because I deliberately want the integer rounding, so even if the user sets the two alt values the same they get a 1m hysterisis, which prevents it oscillating